src/lib/Imp.hs

```diff
@@ -705,18 +747,67 @@ extendIdxsTy (idxsTy, idxs) new = do
 type Idxs n = [AtomName n]
 type IdxNest = Nest Binder
 type DestIdxs n = (EmptyAbs IdxNest n, Idxs n)
+type DepVars n = [AtomName n]
 
 -- TODO: make `DestIdxs` a proper E-kinded thing
 sinkDestIdxs :: DExt n l => DestIdxs n -> DestIdxs l
 sinkDestIdxs (idxsTy, idxs) = (sink idxsTy, map sink idxs)
 
+-- dest for the args and the result
+-- TODO: de-dup some of the plumbing stuff here with the ordinary makeDest path
+type NaryLamDest = Abs (Nest (BinderP AtomNameC Dest)) Dest
+
+makeNaryLamDest :: (ImpBuilder m, EnvReader m)
+                => NaryPiType n -> m n (AbsPtrs NaryLamDest n)
+makeNaryLamDest piTy = liftImmut do
+  DB env <- getDB
+  cache <- get
+  let allocInfo = (LLVM, CPU, Unmanaged) -- TODO! This is just a placeholder
```

Comment:
Can you please at least add a check that we're compiling for the CPU? This should be possible to query from `ImpBuilder`. As is, people will likely get segfaults with the CUDA backend...